File: Hospital Management System/frontend/src/services/doctorService.js
import api from './api';

const doctorService = {
  getAllDoctors: async () => {
    try {
      const response = await api.get('/doctors');
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  addDoctor: async (doctor) => {
    try {
      const response = await api.post('/doctors', doctor);
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  getDoctorById: async (id) => {
    try {
      const response = await api.get(`/doctors/${id}`);
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  updateDoctor: async (id, doctor) => {
    try {
      const response = await api.put(`/doctors/${id}`, doctor);
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  deleteDoctor: async (id) => {
    try {
      const response = await api.delete(`/doctors/${id}`);
      return response.data;
    } catch (error) {
      throw error;
    }
  },
};

export default doctorService;

File: Hospital Management System/backend/src/main/java/com/hospital/management/service/DoctorService.java
package com.hospital.management.service;

import com.hospital.management.model.Doctor;

import java.util.List;
import java.util.Optional;

public interface DoctorService {
    
    Doctor createDoctor(Doctor doctor);
    
    List<Doctor> getAllDoctors();
    
    Optional<Doctor> getDoctorById(Long id);
    
    Doctor updateDoctor(Long id, Doctor doctorDetails);
    
    void deleteDoctor(Long id);
}

